frameworks/helloworld/tests/test_zzzrecovery.py

```diff
@@ -288,18 +288,26 @@ def test_pod_replace():
 # WARNING: THIS MUST BE THE LAST TEST IN THIS FILE. ANY TEST THAT FOLLOWS WILL BE FLAKY.
 # @@@@@@@
 @pytest.mark.sanity
-@pytest.mark.skip(reason="INFINITY-3023")
 def test_shutdown_host():
-    replace_task = sdk_tasks.get_task_avoiding_scheduler(
+    candidate_tasks = sdk_tasks.get_tasks_avoiding_scheduler(
         config.SERVICE_NAME, re.compile('^(hello|world)-[0-9]+-server$'))
-    assert replace_task is not None, 'Could not find a node to shut down'
-    replace_pod_name = replace_task.name[:-len('-server')]
+    assert len(candidate_tasks) != 0, 'Could not find a node to shut down'
+
+    # Pick the host of the first task from the above list, then get ALL tasks which may be located
+    # on that host. We'll need to 'pod replace' all of them.
+    replace_hostname = candidate_tasks[0].host
+    replace_tasks = [
+        task for task in candidate_tasks
+        if task.host == replace_hostname]
+    log.info('Tasks on host {} to be replaced after shutdown: {}'.format(replace_hostname, replace_tasks))
 
     # Instead of partitioning or reconnecting, we shut down the host permanently
-    sdk_cmd.shutdown_agent(replace_task.host)
+    sdk_cmd.shutdown_agent(replace_hostname)
 
-    sdk_cmd.svc_cli(config.PACKAGE_NAME, config.SERVICE_NAME,
-                    'pod replace {}'.format(replace_pod_name))
+    for task in replace_tasks:
+        pod_name = task.name[:-len('-server')]
```

Comment:
What does this do?  Don't we want pod names, not task names?

Reply:
get pod name for `pod replace` operation: `hello-0-server` => `hello-0`

Reply:
I see.  Now I can read it.  Thanks.

Reply:
Should we get a `Set` of pod names before issuing replace calls so we don't call replace on the same pod twice?

Reply:
It shouldn't be a problem here, as there's only 1 task/pod. Could add a sanity check to that effect though.

Reply:
Could use rstrip("-server") instead to make the intent clearer. Also safer if the string doesn't contain "-server"

Reply:
doesn't rstrip just strip anything that matches the set?

```
>>> 'hello'.rstrip('lo')
'he'
```

Reply:
My bad. I have been using it wrong!